### DISCUSSION TITLE
Adding coupling constraint to a coupling model

### DISCUSSION THREAD

**Discussion by Ilhem36** [micom-dev/micom#92] [General]
### Checklist

Hello , I want to know if the microbial model generated by MICOM takes automatically into account the coupling of microbial reaction with their biomass reactions ? 
I want to To enforce that flux through a microbial reaction only occurs when its biomass reaction has a non-zero value, by including coupling constraints. 
But I don't see how can I loop over all microbial reaction to couple each of them with the biomass reactions. 
Do you have any idea please?

**Comment by cdiener** [comment]
> I want to To enforce that flux through a microbial reaction only occurs when its biomass reaction has a non-zero value, by including coupling constraints.

This happens by default in MICOM. If the growth rate is zero, all fluxes in the organism should be zero as well. This is commonly only not the case if your model contains forced demand reactions which MICOM does not allow. 

> But I don't see how can I loop over all microbial reaction to couple each of them with the biomass reactions.

If you really want to couple the bound to growth rate (allowing larger flux bounds if the growth rate is large) then you would do it by exactly doing what you said, by looping over all reactions and coupling them with the biomass of that organism. You can look at the MICOM code base how other algorithms handle this, but it is nothing we support officially right now.

**Comment by Ilhem36** [comment]
Thank for your answer, I do what you said exactly and the coupling constraints are added to the model but when I save the model and download it I can't find these constraints in  the model. 
Do you have please any idea to how to save these constraints in the model ?. Thank you

**Reply by cdiener** [reply]
If you add them and they appear in `model.constraints` it should work when saving and loading as pickle.
